triton-vm/src/table/master_table.rs

```diff
@@ -309,54 +305,95 @@ where
-
         // The following is a batched version of barycentric Lagrangian evaluation.
         // Since the method `barycentric_evaluate` is self-contained, not returning
         // intermediate items necessary for batching, and since returning and reusing
-        // those indermediate items would produce a challenging interface, the relevant
+        // those intermediate items would produce a challenging interface, the relevant
         // parts are reimplemented here.
-        let domain = self.randomized_trace_domain().domain_values();
+
+        let domain = self.trace_domain().domain_values();
         let domain_shift = domain.iter().map(|&d| indeterminate - d).collect();
         let domain_shift_inverses = XFieldElement::batch_inversion(domain_shift);
         let domain_over_domain_shift = domain
             .into_iter()
-            .zip(domain_shift_inverses)
+            .zip_eq(domain_shift_inverses)
             .map(|(d, inv)| d * inv);
-        let denominator_inverse = domain_over_domain_shift
+        let barycentric_eval_denominator_inverse = domain_over_domain_shift
             .clone()
             .sum::<XFieldElement>()
             .inverse();
 
-        Zip::from(ood_row.axis_iter_mut(Axis(0)))
-            .and(self.randomized_trace_table().axis_iter(Axis(1)))
-            .par_for_each(|v, codeword| {
-                let numerator = domain_over_domain_shift
+        let ood_trace_domain_zerofier: XFieldElement =
+            self.trace_domain_zerofier().evaluate(indeterminate);
+
+        let trace_table = self.trace_table();
+        (0..Self::NUM_COLUMNS)
+            .into_par_iter()
+            .map(|i| {
+                let trace_codeword = trace_table.column(i);
+                let barycentric_eval_numerator = domain_over_domain_shift
                     .clone()
-                    .zip(codeword)
+                    .zip_eq(trace_codeword)
                     .map(|(dsi, &abscis)| abscis * dsi)
                     .sum::<XFieldElement>();
 
-                Array0::from_elem((), numerator * denominator_inverse).move_into(v);
-            });
-        ood_row
+                let ood_trace_randomizer: XFieldElement =
+                    self.trace_randomizer_for_column(i).evaluate(indeterminate);
+
+                barycentric_eval_numerator * barycentric_eval_denominator_inverse
+                    + ood_trace_domain_zerofier * ood_trace_randomizer
+            })
+            .collect::<Vec<XFieldElement>>()
+            .into()
+    }
+
+    fn randomized_column_interpolant(&self, idx: usize) -> Polynomial<Self::Field> {
+        let trace_table = self.trace_table();
+        let column_codeword = trace_table.column(idx);
+        let column_interpolant = self
+            .trace_domain()
+            .interpolate(column_codeword.as_slice().unwrap());
+
+        let randomizer = self
+            .trace_randomizer_for_column(idx)
+            .multiply(&self.trace_domain_zerofier());
```

Comment:
My benchmarks indicate that the speedup gained by the implicitly suggested change is indistinguishable from noise, but since this is the performance-critical path of Triton VM, I consider it a worthwhile inclusion regardless.